### PR TITLE
Refactor UIs related to improved token exchange flows

### PR DESCRIPTION
**Pull request by Yasasr1**
### Purpose
This PR includes following changes related to the token exchange grant
- Added a configuration to provide the capability to selectively disable "Linked Accounts" section in the user attributes tab of the application page.
- Refactored the Advance tab of the trusted token issuer connection.
Screenshot 2024-02-28 at 10 46 18 AM

- Refactored the Linked Accounts section in the application user attributes tab.
Screenshot 2024-02-28 at 10 44 51 AM

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/extensions/configs/application.tsx [hunk 1]
```diff
@@ -159,6 +159,19 @@ export const applicationConfig: ApplicationConfig = {
     },
     attributeSettings: {
         advancedAttributeSettings: {
+            isLinkedAccountsEnabled: (templateId: string): boolean => {
```
Comment (pavinduLakshan) [inline apps/console/src/extensions/configs/application.tsx:162]: Let's make sure to send the corresponding PR to the cloud extensions as well.

File: apps/console/src/features/applications/components/settings/attribute-management/advance-attribute-settings.tsx [hunk 2]
```diff
@@ -16,15 +16,13 @@
  * under the License.
  */
 
-import { AlertLevels, TestableComponentInterface } from "@wso2is/core/models";
-import { addAlert } from "@wso2is/core/store";
+import { TestableComponentInterface } from "@wso2is/core/models";
```
Comment (pavinduLakshan) [inline apps/console/src/features/applications/components/settings/attribute-management/advance-attribute-settings.tsx:19]: Let's use `IdentifiableComponentInterface` as `TestableComponentInterface` is now deprecated.

File: apps/console/src/features/applications/components/settings/attribute-management/advance-attribute-settings.tsx [hunk 3]
```diff
@@ -521,80 +518,72 @@ export const AdvanceAttributeSettings: FunctionComponent<AdvanceAttributeSetting
...
-                            hidden={
-                                !applicationConfig.attributeSettings.advancedAttributeSettings
-                                    .showValidateLinkedLocalAccount
-                            }
-                        />)
-                    }
                     <Field.CheckboxLegacy
-                        ariaLabel="Mandate linked local account"
-                        name="mandateLinkedLocalAccount"
+                        listen={ validateLinkedAccountCheckboxHandler }
```
Comment (pavinduLakshan) [inline apps/console/src/features/applications/components/settings/attribute-management/advance-attribute-settings.tsx:538]: Can't we get this checkbox value through the `values` object inside the `submitValues` method and get rid of this custom listener?

File: apps/console/src/features/connections/components/edit/forms/trusted-token-issuer-advanced-configuration-form.tsx [hunk 4]
```diff
@@ -16,20 +16,23 @@
...
-import { AlertLevels, ClaimDialect, ExternalClaim, IdentifiableComponentInterface } from "@wso2is/core/models";
+import { AlertLevels, IdentifiableComponentInterface } from "@wso2is/core/models";
 import { addAlert } from "@wso2is/core/store";
 import { Field, Form } from "@wso2is/form";
-import { Code, Hint, Message, Text } from "@wso2is/react-components";
-import { getDialects, getExternalClaims } from "apps/console/src/features/claims/api";
+import { Hint, Message } from "@wso2is/react-components";
+import { DropdownOptionsInterface } from "apps/console/src/features/applications/components/settings";
+import { getExternalClaims } from "apps/console/src/features/claims/api";
+import { ClaimManagementConstants } from "apps/console/src/features/claims/constants";
```
Comment (pavinduLakshan) [inline apps/console/src/features/connections/components/edit/forms/trusted-token-issuer-advanced-configuration-form.tsx:25]: Let's use the relative paths for imports

File: apps/console/src/features/connections/components/edit/forms/trusted-token-issuer-advanced-configuration-form.tsx [hunk 5]
```diff
@@ -16,20 +16,23 @@
...
 import { IdentityAppsApiException } from "modules/core/dist/types/exceptions";
-import React, { Dispatch, FunctionComponent, ReactElement, SyntheticEvent, useEffect, useState } from "react";
+import { Claim } from "modules/core/src/models";
```
Comment (pavinduLakshan) [inline apps/console/src/features/connections/components/edit/forms/trusted-token-issuer-advanced-configuration-form.tsx:27]:
```suggestion
import { Claim } from "@wso2is/core/src/models";
```

File: apps/console/src/features/applications/components/settings/attribute-management/advance-attribute-settings.tsx (hunk 2 above)
Comment (Yasasr1) [reply apps/console/src/features/applications/components/settings/attribute-management/advance-attribute-settings.tsx]: I need to conditionally disable each checkbox based on the value of the other one (when top one is unchecked bottom one should be disabled and when bottom one is checked top one should be disabled). For this i need to maintain the state of each checkbox

File: apps/console/src/features/applications/components/settings/attribute-management/advance-attribute-settings.tsx (hunk 3 above)
Comment (Yasasr1) [reply apps/console/src/features/applications/components/settings/attribute-management/advance-attribute-settings.tsx:538]: I need to conditionally disable each checkbox based on the value of the other one (when top one is unchecked bottom one should be disabled and when bottom one is checked top one should be disabled). For this i need to maintain the state of each checkbox

**Review by pavinduLakshan — APPROVED** [review]

**Pull request merged**